lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -69,20 +100,43 @@ def footer_track_actions
         FOOTER_TRACK_ACTIONS
       end
 
-      def generate_data_attribute(link, track_action)
+      def generate_data_attribute(link, track_action, ga4_attributes)
+        if ga4_attributes
+          ga4_link = {
+            "event_name": "navigation",
+            "type": "footer",
+            "index": {
+              "index_link": ga4_attributes[:index_link].to_s,
+              "index_section": ga4_attributes[:index_section].to_s,
+              "index_section_count": "5",
+            }, # //Section.link number notation
+            "index-total": @footer_links_count.to_s, # //Across all sections
+            "section": ga4_attributes[:section],
+            # //Use 'Support links', 'Licence' and 'Copyright' for sections 3, 4 & 5
+          }
+        end
+
         {
           track_category: "footerClicked",
           track_action: track_action,
           track_label: link[:href],
           track_options: {
             dimension29: link[:text],
           },
+          ga4_link: ga4_link,
         }
       end
 
-      def add_data_attributes_to_links(items, track_action)
+      def add_data_attributes_to_links(items, track_action, ga4_data)
```

Comment:
This is where all the links get looped through, and then it calls the `generate_data_attribute` and adds the result of that function to the link.